src/mame/machine/gaelco_ds5002fp.h

```diff
@@ -0,0 +1,56 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+#ifndef MAME_MACHINE_GAELCODS5002FP_H
+#define MAME_MACHINE_GAELCODS5002FP_H
+
+#pragma once
+
+#include "cpu/mcs51/mcs51.h"
+#include "machine/nvram.h"
+
+DECLARE_DEVICE_TYPE(GAELCO_DS5002FP,       gaelco_ds5002fp_device)
+DECLARE_DEVICE_TYPE(GAELCO_DS5002FP_WRALLY, gaelco_ds5002fp_wrally_device)
+
+class gaelco_ds5002fp_device_base : public device_t
+{
+public:
+	required_shared_ptr<uint16_t> m_shareram;
+	required_region_ptr<uint8_t> m_mcu_ram;
```

Comment:
These kinds of members really shouldn't be public - it encourages hacks that poke internal stuff directly.